Sources/NameService/Models/Requests/CreateNameRequestParams.swift

```diff
@@ -0,0 +1,50 @@
+import SolanaSwift
+
+struct CreateNameRequestMessage: Codable, BorshSerializable {
+    let owner: String
+    let timestamp: Int64
+
+    func serialize(to writer: inout Data) throws {
+        try owner.serialize(to: &writer)
+        try timestamp.serialize(to: &writer)
+    }
+}
+
+struct CreateNameRequestParams: Codable {
+    let name: String
+    let owner: String
+    let credentials: Credentials
+
+    struct Credentials: Codable {
+        let timestamp: Date
+        let signature: String
+    }
+}
+
+// TODO: Tech debt. Old structure which will be used when user is not authorized with web3auth
+public struct PostParams: Encodable {
+    public init(owner: String, credentials: PostParams.Credentials) {
+        self.owner = owner
+        self.credentials = credentials
+    }
+
+    public let owner: String
+    public let credentials: Credentials
+
+    public struct Credentials: Encodable {
+        public init(geetest_validate: String, geetest_seccode: String, geetest_challenge: String) {
+            self.geetest_validate = geetest_validate
+            self.geetest_seccode = geetest_seccode
+            self.geetest_challenge = geetest_challenge
+        }
+
+        let geetest_validate: String
```

Comment:
Use please `CodingKeys`

Reply:
i don't want to change it because I copied it from previous model. so I need to change it in old screen too (ReserveName.ViewModel). should i do it or can keep it as it is? it will definitely be used and changed later when backed will add logic for captcha